packages/scaffold-ui/test/partials/w3m-swap-details.test.ts

```diff
@@ -0,0 +1,194 @@
+import { elementUpdated, fixture } from '@open-wc/testing'
+import { afterEach, beforeAll, describe, expect, it, vi } from 'vitest'
+
+import { html } from 'lit'
+
+import { NumberUtil } from '@reown/appkit-common'
+import type { CaipNetwork } from '@reown/appkit-common'
+import { ChainController, SwapController } from '@reown/appkit-core'
+import type { SwapTokenWithBalance } from '@reown/appkit-core'
+
+import { WuiSwapDetails } from '../../src/partials/w3m-swap-details'
+import { HelpersUtil } from '../utils/HelpersUtil'
+
+// -- Constants ----------------------------------------- //
+const MOCK_SOURCE_TOKEN = {
+  symbol: 'ETH',
+  decimals: 18,
+  address: 'eip155:1:0x123',
+  name: 'Ethereum',
+  logoUri: 'https://ethereum.org/eth-logo.png',
+  chainId: 1,
+  quantity: {
+    decimals: '18',
+    numeric: '1.0'
+  },
+  price: 2000,
+  value: 2000
+} as SwapTokenWithBalance
+
+const MOCK_TO_TOKEN = {
+  symbol: 'USDC',
+  decimals: 6,
+  address: 'eip155:1:0x456',
+  name: 'USD Coin',
+  logoUri: 'https://usdc.com/logo.png',
+  chainId: 1,
+  quantity: {
+    decimals: '6',
+    numeric: '1000.0'
+  },
+  price: 1,
+  value: 1000
+} as SwapTokenWithBalance
+
+const MOCK_NETWORK: CaipNetwork = {
+  id: 1,
+  name: 'Ethereum',
+  assets: {
+    imageId: 'ethereum',
+    imageUrl: undefined
+  },
+  chainNamespace: 'eip155',
+  caipNetworkId: 'eip155:1',
+  nativeCurrency: {
+    name: 'Ethereum',
+    symbol: 'ETH',
+    decimals: 18
+  },
+  rpcUrls: {
+    default: {
+      http: ['https://eth.llamarpc.com']
+    }
+  }
+}
```

Comment:
https://linear.app/reown/issue/APKT-2275/create-reownappkit-tests-package-to-export-testing-utils